sync/tryrwmutex.go

```diff
@@ -0,0 +1,103 @@
+package sync
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+)
+
+const (
+	readOffset  = uint64(1)
+	writeOffset = uint64(1 << 20)
+	tryOffset   = uint64(1 << 40)
+
+	tryMask = uint64(1099511627775) // equivalent to setting the first 39 bits to '1'.
```

Comment:
Agreed. We may also want to pull the new rand code out into a separate package, and make it slightly more configurable (e.g. allow using any hash, not just blake2b, and allow creating your own `Reader` instance instead of needing to use the global one).